core-rust/mesh-api-server/src/mesh_api/handlers/block.rs

```diff
@@ -0,0 +1,190 @@
+            transaction_identifier,
+        )),
+        operations,
+        related_transactions: None,
+        metadata: None,
+    };
+
+    // see https://docs.cdp.coinbase.com/mesh/docs/models#block
+    let block = models::Block {
+        block_identifier: Box::new(to_mesh_api_block_identifier_from_state_version(
+            state_version,
+        )?),
+        parent_block_identifier: Box::new(to_mesh_api_block_identifier_from_state_version(
+            previous_state_version,
+        )?),
+        timestamp: transaction_identifiers.proposer_timestamp_ms,
+        transactions: vec![transaction],
+        metadata: None,
+    };
+
+    // see https://docs.cdp.coinbase.com/mesh/docs/models#blockresponse
+    Ok(Json(models::BlockResponse {
+        block: Some(Box::new(block)),
+        other_transactions: None,
+    }))
+}
+
+pub(crate) async fn handle_block_transaction(
+    state: State<MeshApiState>,
+    Json(request): Json<models::BlockTransactionRequest>,
+) -> Result<Json<models::BlockTransactionResponse>, ResponseError> {
+    assert_matching_network(&request.network_identifier, &state.network)?;
+
+    let database = state.state_manager.database.snapshot();
+    let mapping_context = MappingContext::new(&state.network);
+
+    let state_version =
+        extract_state_version_from_mesh_api_block_identifier(&request.block_identifier)
+            .map_err(|err| err.into_response_error("block_identifier"))?;
+
+    let transaction_identifiers = database
+        .get_committed_transaction_identifiers(state_version)
+        .ok_or_else(|| {
+            ResponseError::from(ApiError::TransactionNotFound).with_details(format!(
+                "Failed fetching transaction identifiers for state version {}",
+                state_version.number()
+            ))
+        })?;
+
+    let (operations, transaction_identifier) = get_operations_and_transaction_identifier(
+        &mapping_context,
+        database.deref(),
+        state_version,
+        &transaction_identifiers,
+        Some(&request.transaction_identifier.hash),
+    )?;
+
+    // see https://docs.cdp.coinbase.com/mesh/docs/models#transaction
+    let transaction = models::Transaction {
+        transaction_identifier: Box::new(models::TransactionIdentifier::new(
+            transaction_identifier,
+        )),
+        operations,
+        related_transactions: None,
+        metadata: None,
+    };
+
+    // see https://docs.cdp.coinbase.com/mesh/docs/models#blocktransactionresponse
+    Ok(Json(models::BlockTransactionResponse {
+        transaction: Box::new(transaction),
+    }))
+}
+
+fn get_operations_and_transaction_identifier(
+    mapping_context: &MappingContext,
+    database: &StateManagerDatabase<impl ReadableRocks>,
+    state_version: StateVersion,
+    transaction_identifiers: &CommittedTransactionIdentifiers,
+    requested_transaction_identifier: Option<&str>,
+) -> Result<(Vec<models::Operation>, String), ResponseError> {
```

Comment:
Instead of returning a string, let's return a `models::TransactionIdentifier` here. That way we can have a mapping function that creates a `models::TransactionIdentifier` from a state version and act as documentation that a transaction identifier concept is a state version.